Source/DafnyLanguageServer/Language/Symbols/DafnyLangSymbolResolver.cs

```diff
@@ -163,18 +201,41 @@ private class SymbolDeclarationResolver {
           cancellationToken.ThrowIfCancellationRequested();
           methodSymbol.Returns.Add(ProcessFormal(scope, result));
         }
-        ProcessAndRegisterMethodBody(methodSymbol, method.Body);
+        methodSymbol.Block = method.Body == null ? null : ProcessMethodBody(methodSymbol, method.Body);
+        ScopeSymbol? ExpressionHandler(Expression? expression) =>
+          expression == null
+            ? null
+            : ProcessExpression(new ScopeSymbol(methodSymbol, methodSymbol.Declaration), expression);
+        methodSymbol.Ensures = ProcessListAttributedExpressions(method.Ens, ExpressionHandler);
```

Comment:
For the sake of consistency, I'd opt to use this pattern instead:
```cs
functionSymbol.Body.AddRange(ExpressionHandler(function.Body));
```

Reply:
I did not do it for Body since it's not an enumerable, but I did it for `ensures`, `modifies`, `decreases` and `requires`, and same for the functions equivalent. That way, I was able to remove the setter from the fields.